src/main/java/seedu/programmer/ui/CommandBox.java

```diff
@@ -25,6 +25,8 @@
     private final CommandExecutor commandExecutor;
     private final CommandHistory commandHistory = new CommandHistory();
 
+    private boolean lastCommandIsValid = false;
```

Comment:
i think can rename to `isLastCommandValid` ?

Reply:
Would personally prefer writing `if lastCommandIsValid` over `if isLastCommandValid`, but I can rename to be more in line with the module expectations. Good catch 😃

Reply:
I think would be good to follow closely with the module expectations:
How about another suggestion: `wasLastCommandValid`?

I think can also add a comment here to define this boolean flag:
"// boolean flag to keep track if valid command has just been executed and no keyboard press has been executed yet

Reply:
Good idea, I'll add a comment and let's use `isLastCommandValid` for the variable name